src/passes/StringLowering.cpp

```diff
@@ -267,12 +266,15 @@ struct StringLowering : public StringGathering {
       }
     }
 
-    // Add a custom section with the JSON.
-    json << ']';
-    auto str = json.str();
-    auto vec = std::vector<char>(str.begin(), str.end());
-    module->customSections.emplace_back(
-      CustomSection{"string.consts", std::move(vec)});
+    if (json) {
```

Comment:
I double checked and apparently `operator bool()` on stringstreams checks whether there have been any I/O errors, not whether they have any contents. I guess it would be good to add a test demonstrating that no custom section is emitted when there are no strings.

```suggestion
    if (!json.str().empty()) {
```

Reply:
Done